pandas/_libs/tslibs/period.pyx

```diff
@@ -1384,41 +1383,105 @@ cdef class _Period(object):
-        See also
+        See Also
         --------
-        Period.dayofyear : Return the day of the year.
-        Period.daysinmonth : Return the number of days in that month.
+        Period.dayofweek : Day of the week the period lies in.
+        Period.weekday : Alias of Period.dayofweek.
+        Period.day : Day of the month.
+        Period.dayofyear : Day of the year.
 
         Examples
         --------
-        >>> period1 = pd.Period('2012-1-1 19:00', freq='H')
-        >>> period1
-        Period('2012-01-01 19:00', 'H')
-        >>> period1.dayofweek
+        >>> per = pd.Period('2017-12-31 22:00', 'H')
+        >>> per.dayofweek
+        6
+
+        For periods that span over multiple days, the day at the beginning of
+        the period is returned.
+
+        >>> per = pd.Period('2017-12-31 22:00', '4H')
+        >>> per.dayofweek
+        6
+        >>> per.start_time.dayofweek
         6
 
-        >>> period2 = pd.Period('2013-1-9 11:00', freq='H')
-        >>> period2
-        Period('2013-01-09 11:00', 'H')
-        >>> period2.dayofweek
+        For periods with a frequancy higher than days, the last day of the
+        period is returned.
+
+        >>> per = pd.Period('2018-01', 'M')
+        >>> per.dayofweek
+        2
+        >>> per.end_time.dayofweek
         2
         """
         base, mult = get_freq_code(self.freq)
         return pweekday(self.ordinal, base)
 
     @property
     def weekday(self):
+        """
+        Day of the week the period lies in, with Monday=0 and Sunday=6.
+
+        If the period frequency is lower than daily (e.g. hourly), and the
+        period spans over multiple days, the day at the start of the period is
+        used.
+
+        If the frequency is higher than daily (e.g. monthly), the last day
+        of the period is used.
+
+        Returns
+        -------
+        int
+            Day of the week.
+
+        See Also
+        --------
+        Period.dayofweek : Day of the week the period lies in.
+        Period.weekday : Alias of Period.dayofweek.
+        Period.day : Day of the month.
+        Period.dayofyear : Day of the year.
+
+        Examples
+        --------
+        >>> per = pd.Period('2017-12-31 22:00', 'H')
+        >>> per.dayofweek
+        6
+
+        For periods that span over multiple days, the day at the beginning of
+        the period is returned.
+
+        >>> per = pd.Period('2017-12-31 22:00', '4H')
+        >>> per.dayofweek
+        6
+        >>> per.start_time.dayofweek
+        6
+
+        For periods with a frequancy higher than days, the last day of the
+        period is returned.
+
+        >>> per = pd.Period('2018-01', 'M')
+        >>> per.dayofweek
+        2
+        >>> per.end_time.dayofweek
+        2
+        """
+        # Docstring is a duplicate from dayofweek. Reusing docstrings with
+        # Appender doesn't work for properties in Cython files, and setting
+        # the __doc__ attribute is also not possible.
```

Comment:
Would it work using syntax to the effect of `weekday = property(dayofweek.fget, doc=dayofweek.__doc__)`?

Reply:
Thanks for the feedback. That's actually a great idea, didn't know property had an argument for the doc. Unfortunately this doesn't seem to be supported in cython:

```
    weekday = property(dayofweek.fget, doc=dayofweek.__doc__)
                      ^
------------------------------------------------------------

pandas/_libs/tslibs/period.pyx:1430:23: 'dayofweek' is not a constant, variable or function identifier

```